crates/aptos-rosetta/src/construction.rs

```diff
@@ -770,6 +766,136 @@ fn parse_account_transfer_operation(
     Ok(operations)
 }
 
+/// Parses 0x1::primary_fungible_store::transfer(metadata: address, receiver: address, amount: u64)
+fn parse_primary_fa_transfer_operation(
+    server_context: &RosettaContext,
+    sender: AccountAddress,
+    type_args: &[TypeTag],
+    args: &[Vec<u8>],
+) -> ApiResult<Vec<Operation>> {
+    // There should be one type arg
+    if type_args.len() != 1 {
+        return Err(ApiError::TransactionParseError(Some(format!(
+            "Primary fungible store transfer should have one type argument: {:?}",
+            type_args
+        ))));
+    }
+    let mut operations = Vec::new();
+
+    // Retrieve the args for the operations
+    let metadata: AccountAddress = if let Some(metadata) = args.first() {
+        bcs::from_bytes(metadata)?
+    } else {
+        return Err(ApiError::TransactionParseError(Some(
+            "No metadata address in primary fungible transfer".to_string(),
+        )));
+    };
+    let receiver: AccountAddress = if let Some(receiver) = args.get(1) {
+        bcs::from_bytes(receiver)?
+    } else {
+        return Err(ApiError::TransactionParseError(Some(
+            "No receiver address in primary fungible transfer".to_string(),
+        )));
+    };
+    let amount: u64 = if let Some(amount) = args.get(2) {
+        bcs::from_bytes(amount)?
+    } else {
+        return Err(ApiError::TransactionParseError(Some(
+            "No amount in primary fungible transfer".to_string(),
+        )));
+    };
+
+    // Grab currency accordingly
+
+    let maybe_currency = find_fa_currency(&server_context.currencies, metadata);
+
+    if let Some(currency) = maybe_currency {
+        operations.push(Operation::withdraw(
+            0,
```

Comment:
Feels a little error prone. If this is just meant to be index corresponding to when they're pushed to the vec, this could `operations.len()`. Though perhaps that's even more magical and inexplicable hahah. Up to you, I don't for sure know why the Operation needs an operation_index.